adios2@2.8.0 %oneapi build fails: libadios2_fortran.so.2.8.0: version node not found for symbol adios2_adios_init_mod@adios2_adios_init_serial_smod._

### Steps to reproduce the issue

`adios2@2.8.0 %oneapi@2022.1.0` build is failing using:
* `spack/develop` (ec879240396b457f8bf05361a7d55280e0fa19fb from `Fri Jul 29 19:12:02 2022 -0700`)
* Ubuntu 20.04, OneAPI 2022.1.0 (installed via offline installer)
* Docker container image `ecpe4s/ubuntu20.04-runner-x86_64-oneapi:2022-07-01` (OneAPI pre-installed)

Trying to resolve build error for the E4S OneAPI CI stack:
* https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml#L187
* https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml#L217

Concrete spec: [adios2.spec.json.txt](https://github.com/spack/spack/files/9226734/adios2.spec.json.txt)

Concretization:
```
 -   adios2@2.8.0%oneapi@2022.1.0+blosc+bzip2~cuda~dataman~dataspaces+fortran~hdf5~ipo+mpi+pic+png~python+shared+ssc+sst+sz+zfp build_type=Release arch=linux-ubuntu20.04-x86_64
[+]      ^bzip2@1.0.8%oneapi@2022.1.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
[+]          ^diffutils@3.8%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]              ^libiconv@1.16%oneapi@2022.1.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]      ^c-blosc@1.21.1%oneapi@2022.1.0+avx2~ipo build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]          ^cmake@3.23.2%oneapi@2022.1.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
[+]              ^ncurses@6.2%oneapi@2022.1.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
[+]                  ^pkgconf@1.8.0%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]              ^openssl@1.1.1q%oneapi@2022.1.0~docs~shared certs=mozilla patches=3fdcf2d arch=linux-ubuntu20.04-x86_64
[+]                  ^ca-certificates-mozilla@2022-03-29%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^perl@5.34.1%oneapi@2022.1.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
[+]                      ^berkeley-db@18.1.40%oneapi@2022.1.0+cxx~docs+stl patches=b231fcc arch=linux-ubuntu20.04-x86_64
[+]                      ^gdbm@1.19%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^readline@8.1.2%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^zlib@1.2.12%oneapi@2022.1.0+optimize+pic+shared patches=0d38234 arch=linux-ubuntu20.04-x86_64
[+]          ^lz4@1.9.3%oneapi@2022.1.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]          ^snappy@1.1.9%oneapi@2022.1.0~ipo+pic+shared build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]          ^zstd@1.5.2%oneapi@2022.1.0+programs compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]      ^libfabric@1.14.1%oneapi@2022.1.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
[+]      ^libffi@3.4.2%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^libpng@1.6.37%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^mpich@4.0.2%oneapi@2022.1.0~argobots~cuda+fortran+hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath datatype-engine=auto device=ch4 netmod=ofi patches=b4c706e pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]          ^findutils@4.9.0%oneapi@2022.1.0 patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]          ^hwloc@2.7.1%oneapi@2022.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm+shared arch=linux-ubuntu20.04-x86_64
[+]              ^libpciaccess@0.16%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^libtool@2.4.7%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^m4@1.4.19%oneapi@2022.1.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]                          ^libsigsegv@2.13%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^util-macros@1.19.3%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]              ^libxml2@2.9.13%oneapi@2022.1.0~python arch=linux-ubuntu20.04-x86_64
[+]                  ^xz@5.2.5%oneapi@2022.1.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]          ^yaksa@0.2%oneapi@2022.1.0~cuda~rocm arch=linux-ubuntu20.04-x86_64
[+]              ^autoconf@2.69%oneapi@2022.1.0 patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]              ^automake@1.16.5%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]              ^python@3.8.13%oneapi@2022.1.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]                  ^expat@2.4.8%oneapi@2022.1.0+libbsd arch=linux-ubuntu20.04-x86_64
[+]                      ^libbsd@0.11.5%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^libmd@1.0.4%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^gettext@0.21%oneapi@2022.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu20.04-x86_64
[+]                      ^tar@1.34%oneapi@2022.1.0 zip=pigz arch=linux-ubuntu20.04-x86_64
[+]                          ^pigz@2.7%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^sqlite@3.38.5%oneapi@2022.1.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-ubuntu20.04-x86_64
[+]                  ^util-linux-uuid@2.37.4%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^sz@2.1.12.2%oneapi@2022.1.0~fortran~hdf5~ipo~netcdf~pastri~python~random_access+shared~stats~time_compression build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]      ^zfp@0.5.5%oneapi@2022.1.0~aligned~c~cuda~fasthash~fortran~ipo~openmp~profile~python+shared~strided~twoway bsws=64 build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
```

Install error:
```
==> Installing adios2-2.8.0-5nsqk2shsuy5wauk7mkn55soxv3fqcas
==> No binary for adios2-2.8.0-5nsqk2shsuy5wauk7mkn55soxv3fqcas found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/5a/5af3d950e616989133955c2430bd09bcf6bad3a04cf62317b401eaf6e7c2d479.tar.gz
==> No patches needed for adios2
==> adios2: Executing phase: 'cmake'
==> adios2: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

4 errors found in build log:
     1364    [ 90%] Linking CXX shared library ../../lib/libadios2_fortran.so
     1365    /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/cmake-3.23.2-4522tfkc3hkawky76rjynn3nkgkz3u2x/bin/cmake -E touch bindings/Fortran/CMakeFil
             es/adios2_fortran.dir/modules/adios2_mod.f90.o.provides.build
     1366    cd /tmp/root/spack-stage/spack-stage-adios2-2.8.0-5nsqk2shsuy5wauk7mkn55soxv3fqcas/spack-build-5nsqk2s/bindings/Fortran && /spack/opt/spack/linux-ub
             untu20.04-x86_64/oneapi-2022.1.0/cmake-3.23.2-4522tfkc3hkawky76rjynn3nkgkz3u2x/bin/cmake -E cmake_link_script CMakeFiles/adios2_fortran.dir/link.txt
              --verbose=1
     1367    /spack/lib/spack/env/oneapi/icpx -fPIC -O3 -DNDEBUG -shared -Wl,-soname,libadios2_fortran.so.2 -o ../../lib/libadios2_fortran.so.2.8.0 CMakeFiles/ad
             ios2_fortran.dir/f2c/adios2_f2c_adios.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_attribute.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2
             c_io.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_variable.cpp.o CMakeFiles/adios2_fortran.dir/f2c/adios2_f2c_engine.cpp.o CMakeFiles/adios2_f
             ortran.dir/f2c/adios2_f2c_operator.cpp.o CMakeFiles/adios2_fortran.dir/modules/adios2_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_functio
             ns_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_functions_allocate_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_parameters_mod.f
             90.o CMakeFiles/adios2_fortran.dir/modules/adios2_adios_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_adios_init_mod.F90.o CMakeFiles/adios
             2_fortran.dir/modules/adios2_adios_init_serial_smod.F90.o CMakeFiles/adios2_fortran.dir/modules/adios2_attribute_mod.f90.o CMakeFiles/adios2_fortran
             .dir/modules/adios2_attribute_data_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_
             io_open_mod.F90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_open_serial_smod.F90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_define_vari
             able_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_io_define_attribute_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_engine_mod.f9
             0.o CMakeFiles/adios2_fortran.dir/modules/adios2_engine_begin_step_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_engine_put_mod.f90.o CMake
             Files/adios2_fortran.dir/modules/adios2_engine_get_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_variable_mod.f90.o CMakeFiles/adios2_fortr
             an.dir/modules/adios2_variable_min_mod.f90.o CMakeFiles/adios2_fortran.dir/modules/adios2_variable_max_mod.f90.o CMakeFiles/adios2_fortran.dir/modul
             es/adios2_operator_mod.f90.o  -Wl,-rpath,/tmp/root/spack-stage/spack-stage-adios2-2.8.0-5nsqk2shsuy5wauk7mkn55soxv3fqcas/spack-build-5nsqk2s/lib:/sp
             ack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/zfp-0.5.5-4jkvtugjukdaerrz77oaj6qqloafxju3/lib: ../../lib/libadios2_c.so.2.8.0 ../../lib/liba
             dios2_core.so.2.8.0 -pthread -lifport -lifcoremt -lipgo -lpthread -Wl,-rpath-link,/tmp/root/spack-stage/spack-stage-adios2-2.8.0-5nsqk2shsuy5wauk7mk
             n55soxv3fqcas/spack-build-5nsqk2s/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/zfp-0.5.5-4jkvtugjukdaerrz77oaj6qqloafxju3/lib
     1368    /usr/bin/ld: ../../lib/libadios2_fortran.so.2.8.0: version node not found for symbol adios2_adios_init_mod@adios2_adios_init_serial_smod._
     1369    /usr/bin/ld: failed to set dynamic section sizes: bad value
  >> 1370    clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 1371    make[2]: *** [bindings/Fortran/CMakeFiles/adios2_fortran.dir/build.make:512: lib/libadios2_fortran.so.2.8.0] Error 1
     1372    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-adios2-2.8.0-5nsqk2shsuy5wauk7mkn55soxv3fqcas/spack-build-5nsqk2s'
  >> 1373    make[1]: *** [CMakeFiles/Makefile2:1429: bindings/Fortran/CMakeFiles/adios2_fortran.dir/all] Error 2
...
```



### Error message

<details><summary>Error message</summary><pre>
...
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (ec879240396b457f8bf05361a7d55280e0fa19fb)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/9226724/spack-build-out.txt)
[spack-build-01-cmake-out.txt](https://github.com/spack/spack/files/9226723/spack-build-01-cmake-out.txt)

@ax3l @chuckatkins @williamgfc @rscohn2 @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/adios2/package.py
+++ b/var/spack/repos/builtin/packages/adios2/package.py
@@ -91,6 +91,9 @@ class Adios2(CMakePackage, CudaPackage):
     conflicts("%intel@:15")
     conflicts("%pgi@:14")
 
+    # ifx does not support submodules in separate files
+    conflicts("%oneapi@:2022.1.0", when="+fortran")
+
     depends_on("cmake@3.12.0:", type="build")
     depends_on("pkgconfig", type="build")
 

